src/test/java/cz/muni/fi/pa165/TrainerDaoTest.java

```diff
@@ -0,0 +1,312 @@
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+/**
+ * Unit tests to test the {@link TrainerDao}
+ *
+ * @author Matej Poklemba
+ */
+@Transactional
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@ContextConfiguration(classes = KodemonApplicationContext.class)
+public class TrainerDaoTest extends AbstractTestNGSpringContextTests {
+
+    @Inject
+    private PokemonDao pokemonDao;
+
+    @Inject
+    private TrainerDao trainerDao;
+
+    private Trainer trainer;
+    private Pokemon pokemon;
+
+    @BeforeMethod
+    void prepare() {
+        assertThat(trainerDao, is(notNullValue(TrainerDao.class)));
+        trainerDao.deleteAll();
+        assertThat("trainerDao.count() != 0L", trainerDao.count(), is(0L));
+
+        assertThat(pokemonDao, is(notNullValue(PokemonDao.class)));
+        pokemonDao.deleteAll();
+        assertThat("pokemonDao.count() != 0L", pokemonDao.count(), is(0L));
+        preparePokemon();
+        assertThat("pokemon == null", pokemon, is(notNullValue()));
+        pokemonDao.save(pokemon);
+
+        trainer = null;
+    }
+
+    // save(...) tests
+
+    @Test(expectedExceptions = {DataAccessException.class})
+    void testSaveNullTrainer() {
+        trainer = null;
+        assertThat(trainer, is(nullValue(Trainer.class)));
+        trainerDao.save(trainer);
+    }
+
+    @Test
+    void testSaveCorrectInstance() {
+        trainer = randomTrainer(1);
+        trainer.setFirstName("Ash");
+        trainer.setLastName("Ketchum");
+        Date dob = new Calendar.Builder().setDate(1987, 4, 1).build().getTime();
+        trainer.setDateOfBirth(dob);
+        trainerDao.save(trainer);
+        assertThat("gym.getId() == null", trainer.getId(), is(notNullValue(Long.class)));
+        Trainer testTrainer = trainerDao.findOne(trainer.getId());
+        assertThat(testTrainer, is(equalTo(trainer)));
+    }
+
+    @Test
+    void testSaveMultipleCorrectInstances() {
+        int numberOfTrainers = 5;
+        List<Trainer> trainers = randomTrainers(numberOfTrainers);
+        assertThat(trainers.size(), is(equalTo(numberOfTrainers)));
+        trainerDao.save(trainers);
+        assertThat(trainerDao.count(), is(equalTo((long) trainers.size())));
+        assertThat(trainerDao.findAll(), is(equalTo(trainers)));
+    }
+
+    @Test(expectedExceptions = {NullPointerException.class})
+    void testSaveTrainerWithNullFirstName() {
+        trainer = new Trainer();
+        trainer.setLastName("Ketchum");
```

Comment:
Tieto testy na null fieldy by som asi robil tak, ze by som tam vyslovene volal `setField(null)` lebo z tohto, ked to nenastavis to nie je jasne bez toho, aby si sa pozrel do entity na to, ze nema ziadnu default value.